config/environments/production.rb

```diff
@@ -48,7 +48,8 @@
   # mailer host: replace 'example.com' with your root url
   # this will allow links in email text to function correctly
   config.action_mailer.default_url_options =
-    { host: "#{ENV['RAILS_HOST_NAME']}#{ENV['RAILS_RELATIVE_URL_ROOT']}" }
+    { host: ENV['RAILS_HOST_NAME'],
+      script_name: "/#{ENV['RAILS_RELATIVE_URL_ROOT']}" }
```

Comment:
This can just be `ENV['RAILS_RELATIVE_URL_ROOT']`; it works as-is but generates links that look like `example.com//subdirectory/catalog` (note the double-slash).